packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -111,16 +204,45 @@ export class Trigger {
       }
 
       const pushFilter = this.props.gitConfiguration.pushFilter;
+      const pullRequestFilter = this.props.gitConfiguration.pullRequestFilter;
+
+      if (pushFilter?.length && pullRequestFilter?.length) {
+        throw new Error(`cannot specify both pushFilter and pullRequestFilter for the trigger with sourceAction with name '${sourceAction.actionProperties.actionName}'`);
+      }
+      if (!pushFilter?.length && !pullRequestFilter?.length) {
+        throw new Error(`must specify either pushFilter or pullRequestFilter for the trigger with sourceAction with name '${sourceAction.actionProperties.actionName}'`);
+      }
+
       if (pushFilter !== undefined && pushFilter.length > 3) {
         throw new Error(`length of pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' must be less than or equal to 3, got ${pushFilter.length}`);
       }
-
       pushFilter?.forEach(filter => {
         if (filter.tagsExcludes && filter.tagsExcludes.length > 8) {
-          throw new Error(`maximum length of tagsExcludes for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsExcludes.length}`);
+          throw new Error(`maximum length of tagsExcludes in pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsExcludes.length}`);
         }
         if (filter.tagsIncludes && filter.tagsIncludes.length > 8) {
-          throw new Error(`maximum length of tagsIncludes for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsIncludes.length}`);
+          throw new Error(`maximum length of tagsIncludes in pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsIncludes.length}`);
+        }
+      });
+
+      if (pullRequestFilter !== undefined && pullRequestFilter.length > 3) {
+        throw new Error(`length of pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' must be less than or equal to 3, got ${pullRequestFilter.length}`);
+      }
+      pullRequestFilter?.forEach(filter => {
+        if (!filter.branchesExcludes && !filter.branchesIncludes) {
+          throw new Error(`must specify branches in pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}'`);
+        }
```

Comment:
Otherwise the error occurred in CFn.

```
The trigger configuration for the source action is not valid. Make sure to choose valid filter combinations.
```